### ISSUE TITLE
[atlmfc:x86-windows] build failure

### ISSUE THREAD

**Issue by ThrashAbaddon** [microsoft/vcpkg#10460]
**Host Environment**
- OS: Windows 10
Version	10.0.18362 Build 18362
- Compiler: 
```
Microsoft Visual Studio Community 2019
Version 16.5.0
VisualStudio.16.Release/16.5.0+29911.84
Visual C++ for Linux Development   1.0.9.29814
```

**To Reproduce**
`./vcpkg install atlmfc:x86-windows`

**Failure logs** 
-Cut and past the appropriate build messages from the console output.
-Please attach any additional failure logs mentioned in the console output.
```
Error: Building package atlmfc:x86-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: atlmfc:x86-windows
  Vcpkg version: 2020.02.04-nohash
```

**Additional context**
Add any other context about the problem here, such as what you have already tried to resolve the issue.

I tried to install llvm using `.\vcpkg install llvm`.

**Comment by Cheney-W** [comment]
@ThrashAbaddon 
Could you please provide the detailed failure log to us? If it failed in config step, please provide the config-x86-windows-out.log and config-x86-windows-err.log; if it failed in build step, please provide the install-x86-windows-{config}-out.log.

**Comment by ThrashAbaddon** [comment]
@Cheney-W 
Sure thing.

```powershell
PS C:\Users\Kusari\repos\vcpkg> .\vcpkg.exe install atlmfc:x86-windows --debug
[DEBUG] Using vcpkg-root: C:\Users\Kusari\repos\vcpkg
Computing installation plan...
[DEBUG] CreateProcessW("C:\Users\Kusari\repos\vcpkg\downloads\tools\cmake-3.14.0-windows\cmake-3.14.0-win32-x86\bin\cmake.exe" --version)
[DEBUG] cmd_execute_and_stream_data() returned 0 after   164999 us
[DEBUG] CreateProcessW("C:\Users\Kusari\repos\vcpkg\downloads\tools\cmake-3.14.0-windows\cmake-3.14.0-win32-x86\bin\cmake.exe"  -P "C:/Users/Kusari/repos/vcpkg/buildtrees/0.vcpkg_tags.cmake")
[DEBUG] cmd_execute_and_stream_data() returned 0 after    29117 us
The following packages will be built and installed:
    atlmfc[core]:x86-windows
Starting package 1/1: atlmfc:x86-windows
Building package atlmfc[core]:x86-windows...
[DEBUG] <abientries>
[DEBUG] portfile.cmake|5bb6b6c37b23df60e48ef3b1dd4dda265cbbff35
[DEBUG] CONTROL|ed601950514d7e207973214501e2d764b12f9ecb
[DEBUG] cmake|3.14.0
[DEBUG] powershell|ab1effc926b000a6adc12198a1886514ec203621a53b0cd7ec1cd9a8225dccda7e857feaabcfba4004bea73129b986abaad777c4573f44e0af70411226ce08b0
[DEBUG] vcpkg_fixup_cmake_targets|dac334f3bf50b8262be92623168023aa565aa729
[DEBUG] post_build_checks|2
[DEBUG] triplet|e8dc96fd479b53bb0e6748625827c8268017e35a
[DEBUG] features|core
[DEBUG] </abientries>
[DEBUG] CreateProcessW("C:\Program Files (x86)\Microsoft Visual Studio\Installer\vswhere.exe" -all -prerelease -legacy
-products * -format xml)
[DEBUG] cmd_execute_and_stream_data() returned 0 after    31166 us
[DEBUG] Found path: C:\Program Files\Git\cmd\git.exe
[DEBUG] CreateProcessW("C:\Program Files\Git\cmd\git.exe" --version)
[DEBUG] cmd_execute_and_stream_data() returned 0 after    57262 us
[DEBUG] CreateProcessW(cmd /c ""C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Auxiliary\Build\vcvarsall.bat" -vcvars_ver=14.2 amd64_x86   2>&1 <NUL"[DEBUG] cmd_execute_and_stream_data() returned 0 after  3639582 us
[DEBUG] CreateProcessW("C:\Users\Kusari\repos\vcpkg\downloads\tools\cmake-3.14.0-windows\cmake-3.14.0-win32-x86\bin\cmake.exe" "-DCMD=BUILD" "-DPORT=atlmfc" "-DCURRENT_PORT_DIR=C:/Users/Kusari/repos/vcpkg/ports/atlmfc" "-DVCPKG_ROOT_PATH=C:/Users/Kusari/repos/vcpkg" "-DTARGET_TRIPLET=x86-windows" "-DTARGET_TRIPLET_FILE=C:\Users\Kusari\repos\vcpkg\triplets\x86-windows.cmake" "-DVCPKG_PLATFORM_TOOLSET=v142" "-DVCPKG_USE_HEAD_VERSION=0" "-DDOWNLOADS=C:/Users/Kusari/repos/vcpkg/downloads" "-D_VCPKG_NO_DOWNLOADS=0" "-D_VCPKG_DOWNLOAD_TOOL=BUILT_IN" "-DFEATURES=core" "-DALL_FEATURES=" "-DVCPKG_CONCURRENCY=13" "-DGIT=C:/Program Files/Git/cmd/git.exe" -P "C:/Users/Kusari/repos/vcpkg/scripts/ports.cmake")
CMake Error at ports/atlmfc/portfile.cmake:7 (message):
  Unable to locate 'afxres.h'.  Ensure you have installed the ATL/MFC
  component of Visual Studio.
Call Stack (most recent call first):
  scripts/ports.cmake:90 (include)


[DEBUG] cmd_execute_and_stream_data() returned 1 after   264283 us
Error: Building package atlmfc:x86-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: atlmfc:x86-windows
  Vcpkg version: 2020.02.04-nohash

Additionally, attach any relevant sections from the log files above.
[DEBUG] C:\Users\Kusari\repos\vcpkg\toolsrc\src\vcpkg\install.cpp(470)
[DEBUG] Exiting after 5097232 us (5067377 us)
```
Then I tried `git pull` and upgrading everything but still got the error above. Then I tried installing ATL and MFC components in VS2019, you can see selected packages on image bellow
![image](https://user-images.githubusercontent.com/7493180/77621509-e17ce100-6f3c-11ea-80b5-b9678630a53a.png)
 and just in case a complete configuration

```json
{
  "version": "1.0",
  "components": [
    "Microsoft.VisualStudio.Component.CoreEditor",
    "Microsoft.VisualStudio.Workload.CoreEditor",
    "Microsoft.VisualStudio.Component.Roslyn.Compiler",
    "Microsoft.VisualStudio.Component.Roslyn.LanguageServices",
    "Microsoft.Net.Component.4.8.SDK",
    "Microsoft.Net.Component.4.7.2.TargetingPack",
    "Microsoft.Component.MSBuild",
    "Microsoft.VisualStudio.Component.TextTemplating",
    "Microsoft.Net.Component.4.5.2.TargetingPack",
    "Microsoft.Net.Component.4.5.TargetingPack",
    "Microsoft.VisualStudio.Component.IntelliCode",
    "Microsoft.Net.Component.4.TargetingPack",
    "Microsoft.Net.Component.4.5.1.TargetingPack",
    "Microsoft.Net.Component.4.6.TargetingPack",
    "Component.Microsoft.VisualStudio.LiveShare",
    "Microsoft.Net.Component.4.6.2.TargetingPack",
    "Microsoft.VisualStudio.Component.VC.CoreIde",
    "Microsoft.VisualStudio.Component.Windows10SDK",
    "Microsoft.VisualStudio.Component.VC.Tools.x86.x64",
    "Microsoft.VisualStudio.Component.Graphics.Tools",
    "Microsoft.VisualStudio.Component.VC.DiagnosticTools",
    "Microsoft.VisualStudio.Component.Windows10SDK.18362",
    "Component.CPython3.x64",
    "Microsoft.VisualStudio.Component.Debugger.JustInTime",
    "Microsoft.VisualStudio.Component.VC.Redist.14.Latest",
    "Microsoft.VisualStudio.ComponentGroup.NativeDesktop.Core",
    "Microsoft.VisualStudio.ComponentGroup.WebToolsExtensions.CMake",
    "Microsoft.VisualStudio.Component.VC.CMake.Project",
    "Microsoft.VisualStudio.Component.VC.ATL",
    "Microsoft.VisualStudio.Component.VC.ASAN",
    "Microsoft.VisualStudio.Component.VC.Llvm.ClangToolset",
    "Microsoft.VisualStudio.Component.VC.Llvm.Clang",
    "Microsoft.VisualStudio.ComponentGroup.NativeDesktop.Llvm.Clang",
    "Microsoft.VisualStudio.Component.Windows10SDK.17763",
    "Microsoft.VisualStudio.Workload.NativeDesktop",
    "Microsoft.VisualStudio.Component.NuGet.BuildTools",
    "Component.Unreal",
    "Component.MDD.Linux",
    "Component.Linux.CMake",


    "Microsoft.VisualStudio.Component.VC.14.24.ATL",
    "Microsoft.VisualStudio.Component.VC.14.24.ATL.Spectre",
    "Microsoft.VisualStudio.Component.VC.14.24.MFC.Spectre"
  ]
}
```


but in the end I still get the same error

```powershell
PS C:\Users\Kusari\repos\vcpkg> .\vcpkg.exe install atlmfc:x86-windows --debug
[DEBUG] Using vcpkg-root: C:\Users\Kusari\repos\vcpkg
Computing installation plan...
[DEBUG] CreateProcessW("C:\Users\Kusari\repos\vcpkg\downloads\tools\cmake-3.14.0-windows\cmake-3.14.0-win32-x86\bin\cmake.exe" --version)
[DEBUG] cmd_execute_and_stream_data() returned 0 after    29046 us
[DEBUG] CreateProcessW("C:\Users\Kusari\repos\vcpkg\downloads\tools\cmake-3.14.0-windows\cmake-3.14.0-win32-x86\bin\cmake.exe"  -P "C:/Users/Kusari/repos/vcpkg/buildtrees/0.vcpkg_tags.cmake")
[DEBUG] cmd_execute_and_stream_data() returned 0 after    23263 us
The following packages will be built and installed:
    atlmfc[core]:x86-windows
Starting package 1/1: atlmfc:x86-windows
Building package atlmfc[core]:x86-windows...
[DEBUG] <abientries>
[DEBUG] portfile.cmake|5bb6b6c37b23df60e48ef3b1dd4dda265cbbff35
[DEBUG] CONTROL|ed601950514d7e207973214501e2d764b12f9ecb
[DEBUG] cmake|3.14.0
[DEBUG] powershell|ab1effc926b000a6adc12198a1886514ec203621a53b0cd7ec1cd9a8225dccda7e857feaabcfba4004bea73129b986abaad777c4573f44e0af70411226ce08b0
[DEBUG] vcpkg_fixup_cmake_targets|dac334f3bf50b8262be92623168023aa565aa729
[DEBUG] post_build_checks|2
[DEBUG] triplet|e8dc96fd479b53bb0e6748625827c8268017e35a
[DEBUG] features|core
[DEBUG] </abientries>
[DEBUG] CreateProcessW("C:\Program Files (x86)\Microsoft Visual Studio\Installer\vswhere.exe" -all -prerelease -legacy -products * -format xml)
[DEBUG] cmd_execute_and_stream_data() returned 0 after    19747 us
[DEBUG] Found path: C:\Program Files\Git\cmd\git.exe
[DEBUG] CreateProcessW("C:\Program Files\Git\cmd\git.exe" --version)
[DEBUG] cmd_execute_and_stream_data() returned 0 after    20715 us
[DEBUG] CreateProcessW(cmd /c ""C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Auxiliary\Build\vcvarsall.bat" -vcvars_ver=14.2 amd64_x86   2>&1 <NUL"
 & echo cdARN4xjKueKScMy9C6H& set)
[DEBUG] cmd_execute_and_stream_data() returned 0 after  1715489 us
[DEBUG] CreateProcessW("C:\Users\Kusari\repos\vcpkg\downloads\tools\cmake-3.14.0-windows\cmake-3.14.0-win32-x86\bin\cmake.exe" "-DCMD=BUILD" "-DPORT=atlmfc" "-DCURRENT
_PORT_DIR=C:/Users/Kusari/repos/vcpkg/ports/atlmfc" "-DVCPKG_ROOT_PATH=C:/Users/Kusari/repos/vcpkg" "-DTARGET_TRIPLET=x86-windows" "-DTARGET_TRIPLET_FILE=C:\Users\Kusa
ri\repos\vcpkg\triplets\x86-windows.cmake" "-DVCPKG_PLATFORM_TOOLSET=v142" "-DVCPKG_USE_HEAD_VERSION=0" "-DDOWNLOADS=C:/Users/Kusari/repos/vcpkg/downloads" "-D_VCPKG_N
O_DOWNLOADS=0" "-D_VCPKG_DOWNLOAD_TOOL=BUILT_IN" "-DFEATURES=core" "-DALL_FEATURES=" "-DVCPKG_CONCURRENCY=13" "-DGIT=C:/Program Files/Git/cmd/git.exe" -P "C:/Users/Kus
ari/repos/vcpkg/scripts/ports.cmake")
CMake Error at ports/atlmfc/portfile.cmake:7 (message):
  Unable to locate 'afxres.h'.  Ensure you have installed the ATL/MFC
  component of Visual Studio.
Call Stack (most recent call first):
  scripts/ports.cmake:90 (include)


[DEBUG] cmd_execute_and_stream_data() returned 1 after    73691 us
Error: Building package atlmfc:x86-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: atlmfc:x86-windows
  Vcpkg version: 2020.02.04-nohash

Additionally, attach any relevant sections from the log files above.
[DEBUG] C:\Users\Kusari\repos\vcpkg\toolsrc\src\vcpkg\install.cpp(470)
[DEBUG] Exiting after 1950831 us (1943893 us)
```

If you have any ideas what I could try out please let me know. :)

**Comment by Cheney-W** [comment]
You just need install these components:
![image](https://user-images.githubusercontent.com/38240633/77629732-f1201800-6f84-11ea-82fd-d599ab6d884d.png)
Then please check if this folder in your computer:
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.25.28610\atlmfc

If yes, please rerun the **bootstrap-vcpkg.bat** script to generate a new vcpkg.exe and install atlmfc:x86-windows again.

**Comment by ThrashAbaddon** [comment]
@Cheney-W Thank you for detailed response.
After installing only packages on the image I have mentioned folder `C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.25.28610\atlmfc` on my computer.

I tried then `./vcpkg install atlmfc:x86-windows` and this passed successfully.
After that I installed `llvm` using `.\vcpkg install llvm` and this also passed successfully. Then I tried including and compiling something from llvm and this also passed. 

I wanted to get the working software with minimal number of interventions and that is the reason why I skipped running `bootstrap-vcpkg.bat` script. I don't know was that a bad thing but everything seems to work without it.

As for `atlmfc:x86-windows` one could maybe show a more detailed message to the user. Maybe what kind of packages are necessary even without debug mode where it shows that this has something to do ATL/MFC
```
Unable to locate 'afxres.h'.  Ensure you have installed the ATL/MFC
  component of Visual Studio.
```
I bet it isn't easy to find out for current version in VS what is the latest ATL and MFC package that user should look for.
